packages/ember-application/tests/system/initializers_test.js

```diff
@@ -1,422 +1,396 @@
-    run(() => {
-      MyApplication.initializer({ initialize() {} });
+    expectAssertion(() => {
+      run(() => {
+        MyApplication.initializer({ name: 'initializer' });
+      });
     });
-  });
-});
 
-QUnit.test('initializers that throw errors cause the boot promise to reject with the error', function() {
-  QUnit.expect(2);
-  QUnit.stop();
-
-  let MyApplication = Application.extend();
-
-  MyApplication.initializer({
-    name: 'initializer',
-    initialize() { throw new Error('boot failure'); }
-  });
-
-  app = MyApplication.create({
-    autoboot: false
-  });
-
-  try {
-    app.boot().then(
-      (app) => {
-        QUnit.start();
-        ok(false, 'The boot promise should not resolve when there is a boot error');
-      },
-      (err) => {
-        QUnit.start();
-        ok(err instanceof Error, 'The boot promise should reject with an error');
-        equal(err.message, 'boot failure');
-      }
-    );
-  } catch (e) {
-    QUnit.start();
-    ok(false, 'The boot method should not throw');
-    throw e;
+    expectAssertion(() => {
+      run(() => {
+        MyApplication.initializer({ initialize() {} });
+      });
+    });
   }
-});
 
-QUnit.test('initializers are passed an App', function() {
-  let MyApplication = Application.extend();
+  [`@test initializers that throw errors cause the boot promise to reject with the error`](assert) {
+    assert.expect(2);
 
-  MyApplication.initializer({
-    name: 'initializer',
-    initialize(App) {
-      ok(App instanceof Application, 'initialize is passed an Application');
-    }
-  });
+    let MyApplication = Application.extend();
+    let done = assert.async();
 
-  run(() => {
-    app = MyApplication.create({
-      router: false,
-      rootElement: '#qunit-fixture'
+    MyApplication.initializer({
+      name: 'initializer',
+      initialize() { throw new Error('boot failure'); }
     });
-  });
-});
 
-QUnit.test('initializers can be registered in a specified order', function() {
-  let order = [];
-  let MyApplication = Application.extend();
-  MyApplication.initializer({
-    name: 'fourth',
-    after: 'third',
-    initialize(registry) {
-      order.push('fourth');
-    }
-  });
-
-  MyApplication.initializer({
-    name: 'second',
-    after: 'first',
-    before: 'third',
-    initialize(registry) {
-      order.push('second');
-    }
-  });
-
-  MyApplication.initializer({
-    name: 'fifth',
-    after: 'fourth',
-    before: 'sixth',
-    initialize(registry) {
-      order.push('fifth');
-    }
-  });
+    this.runTask(() => {
+      this.createApplication({
+        autoboot: false
+      }, MyApplication);
+    });
 
-  MyApplication.initializer({
-    name: 'first',
-    before: 'second',
-    initialize(registry) {
-      order.push('first');
+    let app = this.application;
+
+    try {
+      app.boot().then(
+        (app) => {
+          assert.ok(false, 'The boot promise should not resolve when there is a boot error');
```

Comment:
`app.boot().then(() => {` is not inside a runloop, so the thennable callback will be called in an autorun making it async. I think you could avoid the async-ness if you wrapped the promise a runloop, it would then flush in the actions queue. Nice catch:

```js
try {
  this.runTask(() => {
    app.boot().then(app => {
      assert.ok(false, 'The boot promise should not resolve when there is a boot error');
    }, error => {
      assert.ok(error instanceof Error, 'The boot promise should reject with an error');
      assert.equal(error.message, 'boot failure');
    });
  });
} catch (error) {
  // ...
```